### PR TITLE
3296 adds a flag for invertible on/off, and decouples transform stack API

### LINKED ISSUE #3296: decouple the transform stack API from `InvertibleTransforms`?
**Is your feature request related to a problem? Please describe.**
Currently the `InvertibleTransforms` is coupled with the transform stack feature, this makes it difficult to create a non-invertible transform at the same time track the applied transforms.

https://github.com/Project-MONAI/MONAI/pull/3295 proposes to have separate interfaces for these concepts.

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

Fixes #3296

### Description
- adds a flag so that we can completely turn off the transform stack
- decouples the APIs for maintaining the transform stack

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by ericspod** [comment]
This looks like a good change but @rijobro should certainly look over it.

**Comment by wyli** [comment]
> This looks like a good change but @rijobro should certainly look over it.

He is on annual leave, could you please help review @Nic-Ma @ericspod? I'm revising this for the upcoming release in November.

File: monai/transforms/inverse.py [hunk 1]
```diff
@@ -63,48 +118,29 @@ class InvertibleTransform(Transform):
...
             return
-        raise RuntimeError("Should inverse most recently applied invertible transform first")
+        raise RuntimeError(f"Error inverting the most recently applied invertible transform {xform_name} {xform_id}.")
+
+    def get_most_recent_transform(self, data: Mapping, key: Hashable = None):
+        """alias of self.peek_transform"""
+        return self.peek_transform(data, key)
 
-    def get_most_recent_transform(self, data: dict, key: Hashable) -> dict:
+    def peek_transform(self, data: Mapping, key: Hashable = None):
```
Comment (Nic-Ma) [inline monai/transforms/inverse.py:136]: I feel we don't need to add this function and use `get_most_recent_transform` as adapter, why not keep using `get_most_recent_transform` for this logic?

Thanks.

File: monai/utils/enums.py [hunk 2]
```diff
@@ -208,7 +211,8 @@ class ForwardMode(Enum):
     EVAL = "eval"
 
 
-class InverseKeys:
+@alias("InverseKeys")
```
Comment (Nic-Ma) [inline monai/utils/enums.py:214]: Maybe we can just deprecate the `InverseKeys` here?

Thanks.

File: monai/transforms/inverse.py (hunk 1 above)
Comment (wyli) [reply monai/transforms/inverse.py:136]: that's to be consistent with pop/push , but I guess it's ok to drop this one

File: monai/utils/enums.py [hunk 3]
```diff
@@ -208,8 +211,27 @@ class ForwardMode(Enum):
...
+    CLASS_NAME = "class"
+    ID = "id"
+    ORIG_SIZE = "orig_size"
+    EXTRA_INFO = "extra_info"
+    DO_TRANSFORM = "do_transforms"
+    KEY_SUFFIX = "_transforms"
+    NONE = "none"
+
+
+@deprecated(since="0.7.0", msg_suffix="use monai.utils.enums.TraceKeys instead.")
```
Comment (Nic-Ma) [inline monai/utils/enums.py:226]: Should it be `0.8.0`?

Thanks.

**Review by Nic-Ma — APPROVED** [review]

**Comment by rijobro** [comment]
Looks great, thanks for implementing this!

**Pull request merged**